prometheus/prometheus_targets.yml

```diff
@@ -1,75 +1,98 @@
 ---
 - labels:
+    nodename: "sourcegraph-docker-compose-host"
```

Comment:
For my culture, where is this value coming from?

Reply:
The metrics used to render the dashboards need to join two pieces of information: 

1. The name of the block device that the particular instance of a service (like gitserver) is using
2. The disk statistics gathered by node-exporter for that block device 

Block devices are not globally unique, multiple nodes can a block each device named `sdb`. Since there can be multiple instances of a given service and node-exporter (1 for each node), we need some way of ensuring that we're only joining the metrics that _come from the same node_ (to ensure that we're referring to the right device). 

That's why we need a new label: `nodename`. This label  represents an identifier for the specific _machine_ that a metric is coming from. For docker-compose we can just hard-code the identifier since there is only one machine. On Kubernetes, our `nodename` comes from the k8s api: https://github.com/sourcegraph/deploy-sourcegraph-helm/blob/2f46069273c75ac64ad3ba32366f355fab87b3c7/charts/sourcegraph/templates/prometheus/prometheus.ConfigMap.yaml#L174-L178

---

You can take a look at https://github.com/sourcegraph/sourcegraph/pull/43768 to see what these Prometheus queries look like (see https://github.com/sourcegraph/sourcegraph/blob/f247e885ffa94cc633bb03946dd1c173f7a6dd9d/monitoring/definitions/shared/diskmetrics.go#L50-L55), or mess around with the following example query (and see the various values of nodename, etc.): 


https://k8s.sgdev.org/-/debug/grafana/goto/5ldFcmvVk